### ISSUE TITLE
OSError: [Errno 8] Exec format error

### ISSUE THREAD

**Issue by zack686** [bpinaya/Udacity-P5-DetectNetCars#2]
Hi,
I am trying to run the CarDetection.py file but I got an Exec format error. Below I have attached a screenshot of the error. Please give me some guidance on resolving this issue. Thanks!
![screenshot from 2017-12-15 00-33-12](https://user-images.githubusercontent.com/19549198/33995318-bcefc53c-e131-11e7-9316-e341b06ef739.png)

**Comment by bpinaya** [comment]
Hi there @zack686 , are you using a virtual machine to run this? It seems you are running it from the jetson, be sure to install moviepy. There is a line that is commented that you should run at least once if you didn't set up moviepy with the correct codecs.
In `CarDetection.py` these lines are commented:
```python
# CarDetection runs Detectnet on a video pipeline (TX1 Tested)

# This might be run only once if no ffmpeg is installed
#import imageio
#imageio.plugins.ffmpeg.download()
```
Change that to:
```python
# CarDetection runs Detectnet on a video pipeline (TX1 Tested)

# This might be run only once if no ffmpeg is installed
import imageio
imageio.plugins.ffmpeg.download()
```
If you run the script for the first time, afterwards you can leave them commented, it downloads the necessary codecs.

**Comment by eduardinjo** [comment]
Hi,
I'm having the same problem even when leaving line uncommented although using TX2:
![image](https://user-images.githubusercontent.com/16897655/43073534-81aa0cf4-8e69-11e8-98bd-3931ffeda539.png)

**Comment by bpinaya** [comment]
Hi there, can you just open python and run those two lines?
```
>>> import imageio
>>> imageio.plugins.ffmpeg.download()
```
And show me the output? I've tested it on the tx2 and it works fine.

**Comment by eduardinjo** [comment]
I found a solution here: https://stackoverflow.com/questions/40004639/moviepy-oserror-exec-format-error-missing-shebang

Which basically set system binary as default to use. Now the problem I'm facing is with caffe installation. Did you use nvcaffe branch and what installation instructions? I'm going to have to reflash jetson anyway because different carrier board but this should only affect ports.

`WARNING: Logging before InitGoogleLogging() is written to STDERR
E0723 11:37:25.361827  4649 common.cpp:114] Cannot create Cublas handle. Cublas won't be available.
^CF0723 11:37:42.631966  4649 cudnn_conv_layer.cpp:53] Check failed: status == CUDNN_STATUS_SUCCESS (4 vs. 0)  CUDNN_STATUS_INTERNAL_ERROR
*** Check failure stack trace: ***
Aborted (core dumped)`

**Comment by bpinaya** [comment]
oh it's that issue with moviepy then, thanks for the catch, I assume all those things were installed from the python side.
I did use nvcaffe but I think that recent versions of caffe should do.
Might just go the safer route and install from nvcaffe.
If you need further help just create another issue, I'll close this one.

**Comment by bpinaya** [comment]
Also note this: https://github.com/bpinaya/DetectNetCars/issues/1
